### ISSUE TITLE
[UPGRADE] Upgrade of pgpool

### ISSUE THREAD

**Issue by erzetpe** [epiphany-platform/epiphany#2075]
**Is your feature request related to a problem? Please describe.**
We need to upgrade pgpool if possible and needed to the newer version. This task should be done after we will upgrade PostgreSQL to v13.

**Describe the solution you'd like**
We want to upgrade if possible and needed to the newer version of pgpool.

**Describe alternatives you've considered**
None.

**Additional context**
This task should be done after upgrade of PostgreSQL to v12.

---

**DoD checklist**

* [x] Changelog updated
* [x] COMPONENTS.md updated / doesn't need to be updated
* [x] Feature has automated tests
* [x] Automated tests passed (QA pipelines)
  * [x] apply
  * [x] upgrade
* [x] Idempotency tested
* [ ] Documentation added / updated / doesn't need to be updated
* [ ] All conversations in PR resolved
* [ ] Solution meets requirements and is done according to design doc
* [ ] Usage compliant with license

**Comment by przemyslavic** [comment]
We are currently using `bitnami/pgpool:4.1.1-debian-10-r29` image.
As for now the latest version is 4.2.2 (`4.2.2-debian-10-r65`).

**Comment by plirglo** [comment]
@przemyslavic i added small configuration change, shouldn't impact you tests: https://github.com/epiphany-platform/epiphany/commit/65742d7032b5783cfe1237f9ea029ffe1d2df613

**Comment by przemyslavic** [comment]
After some time after deployment Pgpools pods crash on some clusters.
```
postgres-pool          pgpool-5478c5d8d4-clq2r                                             1/1     Running            0          18h
postgres-pool          pgpool-5478c5d8d4-pwt74                                             1/1     Running            0          18h
postgres-pool          pgpool-5478c5d8d4-qb969                                             0/1     CrashLoopBackOff   109        18h
```

Messages from different clusters on which the problem was reproduced:
```
    Last State:     Terminated
      Reason:       ContainerCannotRun
      Message:      OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:393: copying bootstrap data to pipe caused: write init-p: broken pipe: unknown
```
```
    Last State:     Terminated
      Reason:       ContainerCannotRun
      Message:      OCI runtime create failed: container_linux.go:380: starting container process caused: process_linux.go:545: container init caused: read init-p: connection reset by peer: unknown
```
```
    Last State:     Terminated
      Reason:       ContainerCannotRun
      Message:      OCI runtime create failed: container_linux.go:380: starting container process caused: container init was OOM-killed (memory limit too low?): unknown
```

**Comment by przemyslavic** [comment]
Fixed in #2545. After increasing the memory limit, the pods ran without crashing.
I checked the maximum memory usage after a few days and for `RHEL/CentOS` it was about 160 MB and for `Ubuntu` about 240 MB.
:heavy_check_mark: new deployments v4.2.4
:heavy_check_mark: upgrade from v4.1.1.
